manticore/core/cpu/abstractcpu.py

```diff
@@ -656,33 +665,42 @@ def decode_instruction(self, pc):
                 if isinstance(c, Constant):
                     c = chr(c.value)
                 else:
-                    logger.error('Concretize executable memory %r %r', c, text )
-                    raise ConcretizeMemory(self.memory, address = pc,
-                                            size = 8 * self.max_instr_width, 
-                                            policy = 'INSTRUCTION' )
+                    logger.error('Concretize executable memory %r %r', c, text)
+                    raise ConcretizeMemory(self.memory,
+                                           address=pc,
+                                           size=8 * self.max_instr_width,
+                                           policy='INSTRUCTION')
             text += c
 
 
         #Pad potentially incomplete intruction with zeroes
 
         code = text.ljust(self.max_instr_width, '\x00')
 
-        #decode the instructtion from code 
         try:
-            instruction = next(self._md.disasm(code, pc))
+            # decode the instruction from code
+            insn = self.disasm.disassemble_instruction(code, pc)
         except StopIteration as e:
-            raise DecodeException(pc, code) 
-
+            raise DecodeException(pc, code)
 
         #Check that the decoded intruction is contained in executable memory
-        if not self.memory.access_ok(slice(pc, pc+instruction.size), 'x'):
+        if not self.memory.access_ok(slice(pc, pc + insn.size), 'x'):
             logger.info("Trying to execute instructions from non-executable memory")
             raise InvalidMemoryAccess(pc, 'x')
 
-        instruction.operands = self._wrap_operands(instruction.operands)
+        # if we are executing Binja-IL but need to fallback to capstone,
```

Comment:
I think we should move this (and the above check for binja-il) to a subclass-specific helper; having this in AbstractCpu feels incorrect.